[Mailer] Document the optional HttpClient dependency (#18387)
symfony/symfony-docs: mailer.rst

    <mail-catcher>` instead. Note that most supported providers also offer a
    free tier.

> .. note::

Reviewer: shouldn't this go with the other notes below the table of schemes ? there, it could explain that all the `https` and `api` transports require the HttpClient.

Author: Good idea!

>     When using a 3rd party transport, you might want to use a web API
>     instead of directly rely on the SMTP protocol to send emails.

Reviewer 2: This sounds a bit like we preferred web APIs over SMTP. I had something more neutral in mind, like…

```suggestion
    Besides SMTP, many 3rd party transports offer a web API to send emails.
```

Author: Applied :+1:

    To do so, you have to install the optional HttpClient dependency

>     thanks to ``composer require symfony/http-client``.

Reviewer 2: We should point out that the HttpClient does not replace the bridge, but in fact needs to be installed in addition to it.

Author: Rephrased, is this better?

        The usage of ``default_socket_timeout`` as the default timeout was
        introduced in Symfony 5.1.

.. note::

    Besides SMTP, many 3rd party transports offer a web API to send emails.
    To do so, you have to install (additionally to the bridge)

>     the Symfony HttpClient with ``composer require symfony/http-client``.

Reviewer 3: ```suggestion
    the HttpClient component via ``composer require symfony/http-client``.
```

Author: Updated!
